### PR TITLE
Add support for accepting wrapped data bag items as provided by knife

### PR DESCRIPTION
**Pull request by seth**
This fixes `knife data bag create BAG1 ITEM1` and still passes pedant tests. Have not added any tests but verified with knife.

### CODE REVIEW

File: src/chef_data_bag_item.erl [hunk 1]
```diff
@@ -92,3 +92,15 @@ validate_data_bag_item(DataBagItem, {update, UrlName}) ->
...
+maybe_unwrap(Ejson) ->
+    case is_wrapped_item(Ejson) of
+        true ->
+            ej:get({<<"raw_data">>}, Ejson);
+        false ->
+            Ejson
+    end.
+
+is_wrapped_item(Ejson) ->
+    (undefined =/= ej:get({<<"raw_data">>}, Ejson) andalso
```
Comment (mmzyk) [inline src/chef_data_bag_item.erl:105]: What is andalso? I've never seen that before and doesn't seem like it is an erlang BIF. Is that something we put in somewhere?

Comment (mmzyk) [inline src/chef_data_bag_item.erl:105]: And apparently it is a BIF. Just found it in the erlang docs. Learn something new everyday.

**Comment by mmzyk** [comment]
I'm going to nitpick a bit - I think the code is fine, but the naming bothers me - I don't think it's as clean as it could be. The function is called maybe_unwrap, which is true in what it does, but it's also very databag specific.

I would suggest renaming maybe_unwrap to unwrap_databag and leaving the internals the same, then renaming is_wrapped_item to is_wrapped_databag. I think that makes the intent more clear.

is_wrapped_item is very databag specific, so call it that. maybe_unwrap, even if named unwrap_databag is always unwrapping the databag - the question just is: does have to take any action to do so and sometimes that is yes and sometimes that is no, but it's still unwrapping the databag.

Long way of saying you might consider renaming the functions.

**Comment by seth** [comment]
Fair points. I have mixed feelings about a more specific name. These are private helper functions in the `chef_data_bag_item` module so name spacing everything to data_bag (not strictly correct, since what is being unwrapped is a data_bag_item) feels a bit overkill. How about `maybe_unwrap_item`? I can see how you could read "item" as generic thing, but I'm thinking of it as short for data bag item.

**Comment by mmzyk** [comment]
Hadn't considered that this is in the data_bag_item module. I could live with maybe_unwrap_item, although I'm not sure the maybe is really needed. Not going to lose sleep over it being named maybe_unwrap_item though.

**Pull request merged**